### PR TITLE
New enip layer

### PR DESCRIPTION
**Pull request by jdiog0**
New EtherNet/IP layer, it needs #1938 first.
Thank you @gpotter2 for #1919

### CODE REVIEW

**Comment by gpotter2** [comment]
This isn't the same thing as https://github.com/secdev/scapy/blob/master/scapy/contrib/etherip.py right ?

I'm a bit confused by EthernetIP and EtherIP :smile:

**Comment by jdiog0** [comment]
> This isn't the same thing as https://github.com/secdev/scapy/blob/master/scapy/contrib/etherip.py right ?
> 
> I'm a bit confused by EthernetIP and EtherIP 😄

No, it is not the same. EtherNet/IP is an Industrial Protocol (https://www.odva.org/Technology-Standards/EtherNet-IP/Overview)

File: scapy/contrib/enipTCP.uts [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+pkt=pkt/ENIPListServicesReply()
+assert(pkt["ENIPListServicesReply"].itemCount == 0)
+
+= ENIP List Services Reply Items Default Values
+pkt=pkt/ENIPListServicesReplyItems()
+assert(pkt["ENIPListServicesReplyItems"].itemTypeCode == 0)
+assert(pkt["ENIPListServicesReplyItems"].itemLength == 0)
+assert(pkt["ENIPListServicesReplyItems"].version == 1)
+assert(pkt["ENIPListServicesReplyItems"].flag == 0)
+assert(pkt["ENIPListServicesReplyItems"].serviceName == None)
```
Comment (gpotter2) [inline scapy/contrib/enipTCP.uts]: The best way is to get the sublayer via the class, rather than the name.

**Comment by gpotter2** [comment]
Sorry for the delay @jdiog0 :/

**Comment by jdiog0** [comment]
> Sorry for the delay @jdiog0 :/

No problem 🙂

**Pull request merged**